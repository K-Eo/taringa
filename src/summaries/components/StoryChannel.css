.StoryChannel {
  color: #6c757d !important;
  font-size: 80%;
  font-weight: 400;
  white-space: nowrap;
  display: inline-block;
  margin-left: 0.25rem;
}
