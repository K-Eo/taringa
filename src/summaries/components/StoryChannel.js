import React from 'react'
import { Link } from 'react-router-dom'

import './StoryChannel.css'

const StoryChannel = ({ details: { channelType, name, title } }) => {
  if (channelType === 'user_feed') {
    return null
  }

  const handleOnClick = e => e.stopPropagation()

  return (
    <div className="StoryChannel">
      <span>en </span>
      <Link to={`/c/${name}`} onClick={handleOnClick}>
        {title}
      </Link>
    </div>
  )
}

StoryChannel.defaultProps = {
  details: {
    channelType: '#CHANNEL_TYPE',
    name: '#CHANNEL_NAME',
    title: '#CHANNEL_TITLE',
  },
}

export default StoryChannel
