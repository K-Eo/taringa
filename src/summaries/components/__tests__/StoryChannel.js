import React from 'react'

import StoryChannel from '../StoryChannel'

describe('StoryChannel', () => {
  it('exists', () => {
    expect(StoryChannel).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<StoryChannel />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders channel details', () => {
    const details = {
      name: 'foo',
      title: 'Foo',
    }

    const wrapper = shallow(<StoryChannel details={details} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('does not render if channel type is user_feed', () => {
    const details = {
      channelType: 'user_feed',
      name: 'foo',
      title: 'Foo',
    }

    const wrapper = shallow(<StoryChannel details={details} />)

    expect(wrapper).toMatchSnapshot()
  })
})
