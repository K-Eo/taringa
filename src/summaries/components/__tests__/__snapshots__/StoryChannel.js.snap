// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryChannel does not render if channel type is user_feed 1`] = `""`;

exports[`StoryChannel renders channel details 1`] = `
<div
  className="StoryChannel"
>
  <span>
    en 
  </span>
  <Link
    onClick={[Function]}
    replace={false}
    to="/c/foo"
  >
    Foo
  </Link>
</div>
`;

exports[`StoryChannel renders default 1`] = `
<div
  className="StoryChannel"
>
  <span>
    en 
  </span>
  <Link
    onClick={[Function]}
    replace={false}
    to="/c/#CHANNEL_NAME"
  >
    #CHANNEL_TITLE
  </Link>
</div>
`;
